File: solutions/1.two-sum.c
// @leet start
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *twoSum(int *nums, int numsSize, int target, int *returnSize) {
  for (int i = 0; i < numsSize; i++) {
    for (int j = 0; j < numsSize; j++) {
      if (nums[i] + nums[j] == target) {
        int *result = malloc(2 * sizeof(int));
        *returnSize = 2;
        result[0] = i;
        result[1] = j;

        return result;
      }
    }
  }
  return 0;
}
// @leet end

File: solutions/1.two-sum.py
# @leet start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for i, num in enumerate(nums):
            rest = target - num
            if rest in cache:
                return [cache[rest], i]

            cache[num] = i


# @leet end

